caas/kubernetes/provider/detectcloud.go

```diff
@@ -27,11 +41,27 @@ func (p kubernetesEnvironProvider) DetectClouds() ([]cloud.Cloud, error) {
 
 // DetectCloud implements environs.CloudDetector.
 func (p kubernetesEnvironProvider) DetectCloud(name string) (cloud.Cloud, error) {
+	k8sConfig, err := clientconfig.GetLocalKubeConfig()
+	if err != nil {
+		return cloud.Cloud{}, errors.Annotate(err, "reading local kubeconf")
+	}
+
+	kubeConfigClouds, err := k8scloud.CloudsFromKubeConfigContexts(k8sConfig)
+	if err != nil {
+		return cloud.Cloud{}, errors.Annotate(err, "parsing kubeconfig to client cloud config")
+	}
+
```

Comment:
This can be combined with the code from DetectClouds() into a helper

Reply:
Done & Done